subsys/bootloader/bl_storage/bl_storage.c

```diff
@@ -99,26 +106,41 @@ static uint32_t check_and_fix_image_trailer(const char* prefix, const mcuboot_im
 		case MCUBOOT_IMAGE_TRAILER_STATUS_INACTIVE:
 			printk("%s: INACTIVE\n\r", prefix);
 			break;
+		case MCUBOOT_IMAGE_TRAILER_STATUS_UNKNOWN:
+			printk("%s: UNKNOWN\n\r", prefix);
+			break;
 		default:
-			printk("%s: Unknown trailer status=0x%X\n\r", trailer->status);
+			printk("%s: UNKNOWN, status=0x%"PRIu32"\n\r", prefix, trailer->status);
 		}
+	} else {
+		printk("%s: UNKNOWN (no image trailer)\n\r", prefix);
 	}
 	return result;
 }
 
 slot_priority slot_priority_from_image_trailers(void)
 {
+	/* preserve behavior of b0 if no magic headers are present, by "prioritizing" S0 as default */
 	int result = SLOT_PRIORITY_S0;
 	uint32_t s0_addr = s0_address_read();
 	uint32_t s1_addr = s1_address_read();
+	/* sanity check that the addresses from "provisioning" and the PM config match */
 	if (s0_addr == PM_S0_OFFSET && s1_addr == PM_S1_OFFSET) {
 		const mcuboot_image_trailer* s0 = (mcuboot_image_trailer*)(s0_addr + PM_S0_SIZE - sizeof(mcuboot_image_trailer));
 		const mcuboot_image_trailer* s1 = (mcuboot_image_trailer*)(s1_addr + PM_S1_SIZE - sizeof(mcuboot_image_trailer));
 		uint32_t s0_status = check_and_fix_image_trailer("slot0", s0);
 		uint32_t s1_status = check_and_fix_image_trailer("slot1", s1);
-		if (s0_status == MCUBOOT_IMAGE_TRAILER_STATUS_INACTIVE || s1_status == MCUBOOT_IMAGE_TRAILER_STATUS_TESTING) {
-			if (s1_status != MCUBOOT_IMAGE_TRAILER_STATUS_INACTIVE)
+		switch (s1_status) {
+		case MCUBOOT_IMAGE_TRAILER_STATUS_TESTING:
+		case MCUBOOT_IMAGE_TRAILER_STATUS_BOOTING:
+		case MCUBOOT_IMAGE_TRAILER_STATUS_ACTIVE:
+			switch (s0_status) {
+			case MCUBOOT_IMAGE_TRAILER_STATUS_TESTING:
+			case MCUBOOT_IMAGE_TRAILER_STATUS_BOOTING:
+				break; /* SLOT_PRIORITY_S0 */
+			default:
 				result = SLOT_PRIORITY_S1;
+			}
```

Comment:
Does the compiler not cry about missing default case here?

Reply:
no - I did not see any warnings related to that, I can add a break but it must be minor

Reply:
Just usually switch statements require all cases to be covered (or have a default) so I figured if we ever change compiler flags to have more warnings or something, this is a low hanging fruit to fix preemptively.
But in practice it's minor, yeah

Reply:
we will keep this - at least for now